.buildbot_patch_yk_dep.py

```diff
@@ -0,0 +1,81 @@
+"""
+Checks the specified pull request description for a special `ci-yk` line and
+patches Config.toml if necessary.
+
+The line should be of the form:
+```
+ci-yk: <github-user> <branch>'
+```
+"""
+
+import sys
+import github3 as gh3
+import subprocess
+
+SOFTDEV_USER = "softdevteam"
+YKRUSTC_REPO = "ykrustc"
+YK_REPO = "yk"
+DEFAULT_BRANCH = "master"
+CARGO_TOML = "Cargo.toml"
+
+
+def get_pr_no():
+    # Get the first line of the commit.
+    proc = subprocess.run(["git", "log", "-1", "--pretty=format:%s"],
```

Comment:
Should we check the process exit code, even just `assert proc.code == 0` or whatever the right incantation is?

Reply:
This is a new(ish) [Python3 thing I found](https://docs.python.org/3/library/subprocess.html#subprocess.run). It's already checked since we pass `check=True`:

> If check is true, and the process exits with a non-zero exit code, a CalledProcessError exception will be raised.

Reply:
Oh!